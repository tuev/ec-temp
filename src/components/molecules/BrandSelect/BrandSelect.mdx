import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './BrandSelect.stories.tsx'
import BrandSelect from './BrandSelect.tsx'

<Story name='basic' />
