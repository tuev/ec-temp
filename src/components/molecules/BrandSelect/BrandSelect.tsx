import React, { useCallback, useState, useMemo, useEffect, FC } from 'react'
import Checkbox from 'components/atoms/Checkbox'
import { FormGroup, FormControlLabel } from '@material-ui/core'
import { keys, get } from 'lodash'

import {
  BrandSelectProps,
  IValueItem,
  GetBrandKey,
  GetBrandLabel,
} from './BrandSelect.types'

const getBrandKey: GetBrandKey = checkedValue => keys(checkedValue)

const getBrandLabel: GetBrandLabel = ({ key, values }) =>
  get(values, [key, 'label'])

export const BrandSelect: FC<BrandSelectProps> = (props: BrandSelectProps) => {
  const { values = {}, onChange } = props
  const [checkedValue, toggleCheck] = useState<IValueItem>({ ...values })
  const handleChange = useCallback(
    (name: string | number) => (event: React.ChangeEvent<HTMLInputElement>) => {
      toggleCheck({
        ...checkedValue,
        [name]: { ...checkedValue[name], value: !!event.target.checked },
      })
    },
    [checkedValue]
  )

  useEffect(() => {
    if (onChange) {
      onChange(checkedValue)
    }
  }, [checkedValue, onChange])

  const valueLabels = useMemo<Array<string | number>>(
    () => getBrandKey(checkedValue),
    [checkedValue]
  )

  return (
    <FormGroup>
      {valueLabels.map(value => (
        <FormControlLabel
          key={value}
          control={
            <Checkbox
              checked={get(checkedValue, [value, 'value'])}
              onChange={handleChange(value)}
              value={value}
              data-testid={`brand-checkbox-${value}`}
            />
          }
          label={getBrandLabel({ key: value, values: checkedValue })}
          data-testid={`brand-label-${value}`}
        />
      ))}
    </FormGroup>
  )
}

export default BrandSelect
