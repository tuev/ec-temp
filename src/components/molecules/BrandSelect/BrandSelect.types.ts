export interface IValueItem {
  [name: string]: {
    value: boolean
    label: string
  }
}

export interface IBrandBasicProps {
  values?: IValueItem
  onChange?: OnChange
}

export type GetBrandKey = (values: IValueItem) => Array<keyof IValueItem>

export type GetBrandLabel = ({
  key,
  values,
}: {
  key: keyof IValueItem
  values: IValueItem
}) => string

export type OnChange = (values: IValueItem) => unknown

export type BrandSelectProps = IBrandBasicProps
