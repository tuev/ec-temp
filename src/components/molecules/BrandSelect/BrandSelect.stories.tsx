import React, { useCallback, useState } from 'react'

import BrandSelect from './BrandSelect'

export default {
  component: BrandSelect,
  includeStories: ['basic'],
  title: 'BrandSelect',
}

export const basic = () => (
  <BrandSelect
    values={{
      brand1: { value: true, label: 'Brand 1' },
      brand2: { value: false, label: 'Brand 2' },
    }}
  />
)
