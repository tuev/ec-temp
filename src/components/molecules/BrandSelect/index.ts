import BrandSelect from './BrandSelect'

export default BrandSelect
