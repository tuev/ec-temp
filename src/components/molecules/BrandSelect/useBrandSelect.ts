import { useCallback, useState, useEffect } from 'react'
import { IBrandValues, UseBrandSelect } from './BrandSelect.types'
import { get } from 'lodash'

const useBrandSelect: UseBrandSelect = (
  initValues: IBrandValues,
  onEffectCallback
) => {
  const [checkedValue, toggleCheck] = useState<IBrandValues>(initValues)
  const handleChange = useCallback(
    (name: string | number) => (
      event: React.ChangeEvent<HTMLInputElement>
    ): unknown =>
      toggleCheck({
        ...checkedValue,
        [name]: {
          ...checkedValue[name],
          value: !!get(event, 'target.checked', false),
        },
      }),
    [checkedValue]
  )
  const setBrandSelect = useCallback(
    values => {
      toggleCheck(values)
    },
    [toggleCheck]
  )

  useEffect(() => {
    if (onEffectCallback) {
      onEffectCallback(checkedValue)
    }
  }, [checkedValue, onEffectCallback, setBrandSelect])

  useEffect(() => {
    setBrandSelect(initValues)
  }, [initValues, setBrandSelect])

  return [checkedValue, handleChange]
}

export default useBrandSelect
