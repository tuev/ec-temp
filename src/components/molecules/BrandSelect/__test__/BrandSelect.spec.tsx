import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import BrandSelect from '..'

describe('App brand select', () => {
  it('Test get default brand select no params ', () => {
    // test snapshot
    const wrapper = render(<BrandSelect />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default brand select ', () => {
    const values = {
      brand1: { value: true, label: 'Brand 1' },
      brand2: { value: false, label: 'Brand 2' },
    }
    // test snapshot
    const wrapper = render(<BrandSelect values={values} />)
    expect(wrapper).toMatchSnapshot()
    const brandSlug = Object.keys(values)
    const { container } = wrapper
    const brandSelection = container.querySelectorAll('input')
    expect(brandSelection.length).toEqual(Object.keys(values).length)
    brandSlug.forEach((slug, idx) => {
      const label = wrapper.getByTestId(`brand-label-${slug}`)
      expect(label).toHaveTextContent(values[slug].label)
      expect(brandSelection[idx].checked).toEqual(values[slug].value)
    })
  })

  it('Test onChange brand select ', () => {
    const values = {
      brand1: { value: true, label: 'Brand 1' },
      brand2: { value: false, label: 'Brand 2' },
    }
    // test snapshot
    const onChange = values => console.log('brand-select: values', values)
    const wrapper = render(<BrandSelect values={values} onChange={onChange} />)
    expect(wrapper).toMatchSnapshot()
    const brandSlug = ['brand1', 'brand2']
    const { container } = wrapper
    const brandSelection = container.querySelectorAll('input')
    brandSlug.forEach((slug, idx) => {
      const label = wrapper.getByTestId(`brand-label-${slug}`)
      // expect(label).toHaveTextContent(values[slug].label)
      expect(brandSelection[idx].checked).toEqual(values[slug].value)
      fireEvent.click(label)
      expect(brandSelection[idx].checked).toEqual(!values[slug].value)
    })
  })
})
