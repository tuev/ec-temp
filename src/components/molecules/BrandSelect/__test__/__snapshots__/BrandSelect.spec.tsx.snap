// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App brand select Test get default brand select  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="brand-label-brand1"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-18 Mui-checked MuiIconButton-colorPrimary"
            data-testid="brand-checkbox-brand1"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-20"
                data-indeterminate="false"
                type="checkbox"
                value="brand1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Brand 1
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
          data-testid="brand-label-brand2"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            data-testid="brand-checkbox-brand2"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-20"
                data-indeterminate="false"
                type="checkbox"
                value="brand2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Brand 2
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
        data-testid="brand-label-brand1"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary sc-AykKC KUWBx PrivateSwitchBase-checked-18 Mui-checked MuiIconButton-colorPrimary"
          data-testid="brand-checkbox-brand1"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-20"
              data-indeterminate="false"
              type="checkbox"
              value="brand1"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Brand 1
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
        data-testid="brand-label-brand2"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary sc-AykKC KUWBx MuiIconButton-colorPrimary"
          data-testid="brand-checkbox-brand2"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-20"
              data-indeterminate="false"
              type="checkbox"
              value="brand2"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Brand 2
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App brand select Test get default brand select no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App brand select Test onChange brand select  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="brand-label-brand1"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-97 Mui-checked MuiIconButton-colorPrimary"
            data-testid="brand-checkbox-brand1"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-99"
                data-indeterminate="false"
                type="checkbox"
                value="brand1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Brand 1
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
          data-testid="brand-label-brand2"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            data-testid="brand-checkbox-brand2"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-99"
                data-indeterminate="false"
                type="checkbox"
                value="brand2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Brand 2
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
        data-testid="brand-label-brand1"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiCheckbox-root MuiCheckbox-colorPrimary sc-AykKC KUWBx PrivateSwitchBase-checked-97 Mui-checked MuiIconButton-colorPrimary"
          data-testid="brand-checkbox-brand1"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-99"
              data-indeterminate="false"
              type="checkbox"
              value="brand1"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Brand 1
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
        data-testid="brand-label-brand2"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiCheckbox-root MuiCheckbox-colorPrimary sc-AykKC KUWBx MuiIconButton-colorPrimary"
          data-testid="brand-checkbox-brand2"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-99"
              data-indeterminate="false"
              type="checkbox"
              value="brand2"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Brand 2
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
